en/tutorials/rag-blueprint.md

```diff
@@ -0,0 +1,1475 @@
+
+```txt
+field chunk_embeddings type tensor<bfloat16>(chunk{}, x) {
+  indexing: input text | chunk fixed-length 1024 | embed | attribute 
+  attribute: paged
+}
+```
+
+For example, if you want to calculate `closeness` for a paged embedding vector in first-phase, consider configuring your retrieval operators (typically `weakAnd` and/or `nearestNeighbor`, optionally combined with filters) so that not too many hits are matched. Another option is to enable match-phase limiting, see [match-phase docs](../reference/schema-reference.html#match-phase). In essence, you restrict the number of matches by specifying an attribute field.
+
+## Consider float-binary for ranking
+
+In our blueprint, we choose to index binary vectors of the documents. This does not prevent us from using the float-representation of the query embedding though.
+
+By unpacking the binary document chunk embeddings to their float representations (using [`unpack_bits`](../reference/ranking-expressions.html#unpack-bits)), we can calculate the similarity between query and document with slightly higher precision using a `float-binary` dot product, instead of hamming distance (`binary-binary`)
+
+Below, you can see how we can do this:
+
+```txt
+rank-profile collect-training-data {
+ 
+        inputs {
+            query(embedding) tensor<int8>(x[96])
+            query(float_embedding) tensor<float>(x[768])
+        }
+        
+        function chunk_emb_vecs() {
+            expression: unpack_bits(attribute(chunk_embeddings))
+        }
+
+        function chunk_dot_prod() {
+            expression: reduce(query(float_embedding) * chunk_emb_vecs(), sum, x)
+        }
+
+        function vector_norms(t) {
+            expression: sqrt(sum(pow(t, 2), x))
+        }
+        function chunk_sim_scores() {
+            expression: chunk_dot_prod() / (vector_norms(chunk_emb_vecs()) * vector_norms(query(float_embedding)))
+        }
+
+        function top_3_chunk_text_scores() {
+            expression: top(3, chunk_text_scores())
+        }
+
+        function top_3_chunk_sim_scores() {
+            expression: top(3, chunk_sim_scores())
+        }
+}
+```
+
+### Use complex linguistics/recall only for precision
+
+TODO: this section needs to be improved, if it should be added...
+
+Vespa gives you extensive control over linguistics: decide match mode, stemming, normalization, or control generated tokens.
+
+…and token-based recall.
+Use more specific operators than WeakAND to match only close occurrences `{near}`, multiple alternatives `{equiv}`, weight items, set connectivity, and apply query-rewrite rules.
+
+Don’t use this to increase recall — improve your embedding model instead.
```

Comment:
I think this line is super-useful and it should be in bold black letters (I'm quoting Ross from Friends here). So please keep this section :) Let me know if you need help polishing it.